### PR TITLE
Initial Collaborative Document editing support

### PR DESCRIPTION
**Pull request by haiodo**
Signed-off-by: Andrey Sobolev <haiodo@gmail.com>

# Contribution checklist

## Brief description

Collaborative workflow with documents.

Screenshot 2022-10-20 at 18 02 32
Screenshot 2022-10-20 at 18 02 49

## Checklist

* [ ] - UI test added to added/changed functionality?
* [x] - Screenshot is added to PR if applicable ?
* [x] - Does the code work? Check whether function and logic are correct.
* [x] - Does Changelog.md is updated with changes?
* [x] - Does the translations are up to date?
* [x] - Does it well tested?
* [x] - Tested for Chrome.
* [x] - Tested for Safari.
* [x] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [x] - Rebase your branch onto master and upstream branch
* [x] - Is there any redundant or duplicate code?
* [x] - Are required links are linked to PR?
* [x] - Does new code is well documented ?

## Related issues

A list of closed updated issues

### CODE REVIEW

File: plugins/document-assets/lang/ru.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "string": {
+    "CreateEmployee": "Создать документ"
```
Comment (BykhovDenis) [inline plugins/document-assets/lang/ru.json]: Please fix it and add tests

File: models/document/src/migration.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+async function createSpace (tx: TxOperations): Promise<void> {
+  const contacts = await tx.findOne(core.class.Space, {
+    _id: document.space.Documents
+  })
+  if (contacts === undefined) {
+    await tx.createDoc(
+      core.class.Space,
+      core.space.Space,
+      {
+        name: 'Contacts',
```
Comment (BykhovDenis) [inline models/document/src/migration.ts]: Please rename it

File: models/document/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
+//
+// Copyright © 2020, 2021 Anticrm Platform Contributors.
```
Comment (BykhovDenis) [inline models/document/src/index.ts]: We need way to fix it everywhere (or probably do it manually)

File: packages/core/src/operations.ts [hunk 4]
```diff
@@ -203,4 +191,40 @@ export class TxOperations implements Omit<Client, 'notify'> {
     }
     return this.removeDoc(doc._class, doc.space, doc._id)
   }
+
+  apply (scope: string): ApplyOperations {
+    return new ApplyOperations(this, scope)
+  }
+}
+
+export class ApplyOperations extends TxOperations {
```
Comment (BykhovDenis) [inline packages/core/src/operations.ts:207]: Missing release tag

File: plugins/document/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  stepType: string
+  from: number
+  to: number
+  slice: any
+}
+
+/**
+ * Document in transient state describe a list of editing documents.
+ */
+export interface RichDocumentContent extends AttachedDoc {
```
Comment (BykhovDenis) [inline plugins/document/src/index.ts:36]: Missing release tag

**Review by BykhovDenis — APPROVED** [review]

File: models/document/src/migration.ts (hunk 2 above)
Comment (haiodo) [reply models/document/src/migration.ts]: Fixed.

File: models/document/src/index.ts (hunk 3 above)
Comment (haiodo) [reply models/document/src/index.ts]: Fixed.

File: packages/core/src/operations.ts (hunk 4 above)
Comment (haiodo) [reply packages/core/src/operations.ts:207]: Fixed.

File: plugins/document/src/index.ts (hunk 5 above)
Comment (haiodo) [reply plugins/document/src/index.ts:36]: Fixed.

**Pull request merged**